### PR TITLE
Memberships: Fix issue with tier resolver

### PR DESCRIPTION
**Pull request by millerf**
See p1699963507382499/1699898012.697669-slack-C052XEUUBL4
## Proposed changes:

* Rename getNewsletterProducts to getNewsletterTierProducts to clear confusion
* Fix resolver, which would not filter the right products

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Apply this on your sandbox
* Sandbox ericktesting1571.wordpress.com
* visit https://ericktesting1571.wordpress.com/ admin
* You should see the right plans in the editor 
And this should NOT happen anymore:
Screenshot 2023-11-14 at 13 15 15

### CODE REVIEW

File: projects/plugins/jetpack/extensions/store/membership-products/resolvers.js [hunk 1]
```diff
@@ -151,13 +151,15 @@ const setDefaultProductIfNeeded = ( selectedProductId, setSelectedProductId, sel
...
-export const getNewsletterProducts = (
+export const getNewsletterTierProducts = (
 	productType = PRODUCT_TYPE_PAYMENT_PLAN,
 	selectedProductId = 0,
 	setSelectedProductId = () => {}
 ) =>
 	// Returns the products, but silences the snack bar if a default product is created
-	getProducts( productType, selectedProductId, setSelectedProductId, false );
+	getProducts( productType, selectedProductId, setSelectedProductId, false ).filter(
+		product => product.type === TYPE_TIER
```
Comment (lezama) [inline projects/plugins/jetpack/extensions/store/membership-products/resolvers.js]: they will always have a `type`?

Comment (millerf) [reply projects/plugins/jetpack/extensions/store/membership-products/resolvers.js]: Tiers? Yes. It is explained p81Rsd-1wJ-p2

File: projects/plugins/jetpack/extensions/shared/memberships/settings.js [hunk 2]
```diff
@@ -80,9 +80,9 @@ export function useSetTier() {
 
 function TierSelector() {
 	// TODO: figure out how to handle different currencies
-	const products = useSelect( select => select( membershipProductsStore ).getProducts() )
-		.filter( product => product.subscribe_as_site_subscriber && product.interval === '1 month' )
-		.sort( ( p1, p2 ) => Number( p2.price ) - Number( p1.price ) );
```
Comment (n3f) [inline projects/plugins/jetpack/extensions/shared/memberships/settings.js]: I don't see equivalent logic anywhere.  Do you limit to just the months to determine the label or is there a different mechanism I'm missing?

Comment (millerf) [reply projects/plugins/jetpack/extensions/shared/memberships/settings.js]: You're right.. Nice catch!

**Comment by chrismccluskey** [comment]
I think branch needs a trunk merge. I'm not able to get the paywall block to function properly (post id 110, as shown in your screenshot), and I suspect this could be the reason why.

File: projects/plugins/jetpack/extensions/shared/memberships/settings.js (hunk 2 above)
Comment (millerf) [reply projects/plugins/jetpack/extensions/shared/memberships/settings.js]: Fixed in https://github.com/Automattic/jetpack/pull/34107/commits/884972b3bdc62f7fb434e6d93923ec08fceac1b1

**Comment by millerf** [comment]
@chrismccluskey I updated with trunk
I can see the plan on https://wordpress.com/post/ericktesting1571.wordpress.com/110


Screenshot 2023-11-21 at 13 25 09

**Comment by chrisfromthelc** [comment]
I had the same issues as @chrismccluskey . For one, the paywall block appears broken for me. Even outside of that, plans never load in the sidebar. I added myself to the site as an admin thinking that may have been it (I saw that @millerf and @n3f were), but that didn't seem to help.

I also tried creating a new post with a paywall block, and had the same broken block issue.

**Comment by millerf** [comment]
@chrisfromthelc @chrismccluskey 
Did you sandbox the site you were using the editor? If you were going though Calypso (wordpress.com), that patch would not apply...

I am trying to understand what is going on with your tests....

**Comment by millerf** [comment]
I could reproduce. I have no idea what the issue is. Relaunching a build to test.

Screenshot 2023-11-23 at 16 03 50

**Comment by millerf** [comment]
I fixed it. It is a mere renaming, but  things work...

**Comment by chrismccluskey** [comment]
The issue with the broken block is fixed. However, I'm not seeing any plans on the right hand side. I'm not sure if this is intentional, but since it doesn't match up with one of your previous screenshots I thought I'd mention.

Screenshot 2023-11-23 at 12 56 53 PM

> Did you sandbox the site you were using the editor? If you were going though Calypso (wordpress.com), that patch would not apply...

I've directed `ericktesting1571.wordpress.com` to my sandbox. `subscribe.wordpress.com` and `public-api.wordpress.com` are also pointed to my sandbox. I tried editing the post using the `/wp-admin/` interface, using the gutenberg editor. I'm going through `ericktesting1571.wordpress.com/wp-admin/`, not `wordpress.com` directly.

I can see that I'm getting the products via public-api (as confirmed looking through dev tools networking, searching for "memberships" and looking at the requests). I can also toggle between production and test via sandbox controls and see different products sent. 

On sandbox, this branch is applied directly to trunk.

**Comment by millerf** [comment]
@chrismccluskey 

The site has no tier on sandbox store:

```
~/public_html (trunk) $ wp memberships get-products  https://ericktesting1571.wordpress.com/
Environment: SANDBOX MODE ENABLED - Viewing test data

Site ID:     221980818
Site Domain: ericktesting1571.wordpress.com
Site URL:    https://ericktesting1571.wordpress.com/

---

Connected Account Product ID: 2270
Interval:                     1 month
Price:                        1.00
Currency:                     USD
Multiple Per User:            
Subscribe as Site Subscriber: 
Welcome Email Content:        
Is Editable:                  1
Is Deleted:                   
Title:                        Monthly Subscription
Site ID:                      221980818
Buyer Can Change Amount:      
Type:                         
Tier:                         
ID:                           19
Description:                  

---

Connected Account Product ID: 2273
Interval:                     1 year
Price:                        1.00
Currency:                     USD
Multiple Per User:            
Subscribe as Site Subscriber: 
Welcome Email Content:        
Is Editable:                  1
Is Deleted:                   
Title:                        Yearly Subscription adsf
Site ID:                      221980818
Buyer Can Change Amount:      
Type:                         
Tier:                         
ID:                           22
Description:                  

---


```

**Comment by millerf** [comment]
And on production there is only one tier:
```

wpdev@millerf01.dev.dfw.wordpress.com:~/public_html (trunk) $ wp memberships get-products  https://ericktesting1571.wordpress.com/ --type=tier
Environment: PRODUCTION - Viewing live data

Site ID:     221980818
Site Domain: ericktesting1571.wordpress.com
Site URL:    https://ericktesting1571.wordpress.com/

---

Connected Account Product ID: 329021
Interval:                     1 year
Price:                        6.00
Currency:                     USD
Multiple Per User:            
Subscribe as Site Subscriber: 
Welcome Email Content:        
Type:                         tier
Tier:                         112
Is Editable:                  1
Is Deleted:                   
Title:                        Newsletter (11-13-2023) (yearly)
Site ID:                      221980818
Buyer Can Change Amount:      
ID:                           113
Description:                  

---

Connected Account Product ID: 329020
Interval:                     1 month
Price:                        0.50
Currency:                     USD
Multiple Per User:            
Subscribe as Site Subscriber: 
Welcome Email Content:        
Type:                         tier
Tier:                         
Is Editable:                  1
Is Deleted:                   
Title:                        Newsletter (11-13-2023)
Site ID:                      221980818
Buyer Can Change Amount:      
ID:                           112
Description:                  

---
```

So the selector would not show it anyway, unless a new one is added...

**Comment by millerf** [comment]
I think there was a small issue that is now fixed, but I would like to ship this for naming purposes now.

**Review by chrismccluskey — APPROVED** [review]
LGTM!

**Pull request merged**